### PR TITLE
feat: add new text/plain strategy and use with original object

### PR DESCRIPTION
**Pull request by MiguelAHM**

### CODE REVIEW

File: whois-api/src/test/java/net/ripe/db/whois/api/rest/WhoisSearchServiceTestIntegration.java [hunk 1]
```diff
@@ -216,6 +214,28 @@ public void search_accept_json() {
...
+                "tech-c:         TP1-TEST\n" +
+                "mnt-by:         OWNER-MNT\n" +
+                "source:         TEST\n" +
+                "created:        2017-05-16T11:18:05Z\n" +
+                "last-modified:  2017-05-16T11:18:05Z"));
+        final String whoisResources = RestTest.target(getPort(), "whois/search?query-string=AS102&source=TEST")
+                .request(MediaType.TEXT_PLAIN)
+                .get(String.class);
+
+        //assertThat(whoisResources.getErrorMessages(), is(empty()));
```
Comment (eshryane) [inline whois-api/src/test/java/net/ripe/db/whois/api/rest/WhoisSearchServiceTestIntegration.java:233]: Add some asserts 😉  (they're commented out)

File: whois-api/src/test/java/net/ripe/db/whois/api/rest/WhoisSearchServiceTestIntegration.java [hunk 2]
```diff
@@ -216,6 +214,28 @@ public void search_accept_json() {
         assertThat(whoisObject.getPrimaryKey().get(0).getValue(), is("TP1-TEST"));
     }
 
+    @Test
+    public void search_accept_text_plain() {
```
Comment (eshryane) [inline whois-api/src/test/java/net/ripe/db/whois/api/rest/WhoisSearchServiceTestIntegration.java:218]: Please add tests with multiple objects in the response (e.g. an inetnum and a role), including comment(s)

Comment (eshryane) [reply whois-api/src/test/java/net/ripe/db/whois/api/rest/WhoisSearchServiceTestIntegration.java:218]: Also add a test for a lookup of a non-existant object, what's in the response body? Does it include `%ERROR:101: no entries found` ?

File: whois-api/src/test/java/net/ripe/db/whois/api/rest/WhoisSearchServiceTestIntegration.java (hunk 1 above)
Comment (MiguelAHM) [reply whois-api/src/test/java/net/ripe/db/whois/api/rest/WhoisSearchServiceTestIntegration.java]: Yes, I created the PRs without the tests, this part of code is just to help me to see the output

File: whois-api/src/test/java/net/ripe/db/whois/api/util/TestUtil.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
+package net.ripe.db.whois.api.util;
+
+
+import net.ripe.db.whois.common.rpsl.RpslObject;
+
+public class TestUtil {
+    public static final String AS102_STRING = "" +
```
Comment (eshryane) [inline whois-api/src/test/java/net/ripe/db/whois/api/util/TestUtil.java:7]: Please don't move the test data out of the test class, this is less readable. I'm fine with some duplication if it's more readable, and can be modified in a test class without side-effects elsewhere.

File: whois-api/src/test/java/net/ripe/db/whois/api/rest/WhoisSearchServiceTestIntegration.java (hunk 2 above)
Comment (eshryane) [reply whois-api/src/test/java/net/ripe/db/whois/api/rest/WhoisSearchServiceTestIntegration.java:221]: Thanks for adding tests

File: whois-api/src/main/java/net/ripe/db/whois/api/rest/marshal/AbstractStreamingMarshalCommon.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+import net.ripe.db.whois.api.rest.RestServiceHelper;
+import net.ripe.db.whois.common.Message;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.Response;
+import java.util.List;
+
+public class AbstractStreamingMarshalCommon implements  StreamingMarshal{
```
Comment (eshryane) [inline whois-api/src/main/java/net/ripe/db/whois/api/rest/marshal/AbstractStreamingMarshalCommon.java:11]: Remove the word `Abstract` from the class, it's not `abstract`.

Comment (eshryane) [inline whois-api/src/main/java/net/ripe/db/whois/api/rest/marshal/AbstractStreamingMarshalCommon.java:11]: Also consider combining this class with `StreamingMarshal` to simplify the hierarchy (i.e. can they be a single class?)

File: whois-api/src/main/java/net/ripe/db/whois/api/rest/RpslObjectStreamer.java [hunk 5]
```diff
@@ -200,7 +202,11 @@ private void streamObject(@Nullable final RpslObject rpslObject) {
                 whoisObjectServerMapper.mapManagedAttributes(whoisObject, parameters, rpslObject);
                 whoisObjectServerMapper.mapResourceHolder(whoisObject, parameters, rpslObject);
 
-                streamingMarshal.writeArray(whoisObject);
+                if(streamingMarshal instanceof StreamingMarshalTextPlain){
```
Comment (eshryane) [inline whois-api/src/main/java/net/ripe/db/whois/api/rest/RpslObjectStreamer.java:205]: Add spaces for readability and consistent formatting (they're free! :) )

File: whois-api/src/main/java/net/ripe/db/whois/api/rest/RestServiceHelper.java [hunk 6]
```diff
@@ -125,6 +127,21 @@ public static WhoisResources createErrorEntity(final HttpServletRequest request,
         return whoisResources;
     }
 
+    public static String createErrorStringEntity(final HttpServletRequest request,
```
Comment (eshryane) [inline whois-api/src/main/java/net/ripe/db/whois/api/rest/RestServiceHelper.java:130]: Can you move this into the (Common) StreamingMarshal class? It's only used by the StreamingMarshal implementations. We should reduce the use of 'utility' code as much as possible.

File: whois-api/src/main/java/net/ripe/db/whois/api/rest/marshal/AbstractStreamingMarshalCommon.java [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+        // deliberately not implemented
+    }
+
+    @Override
+    public <T> void endArray() {
+        // deliberately not implemented
+    }
+
+    @Override
+    public <T> void returnCustomError(HttpServletRequest request, List<Message> errorMessages) {
```
[the remark this answers is not in the document]
Comment (eshryane) [reply whois-api/src/main/java/net/ripe/db/whois/api/rest/marshal/AbstractStreamingMarshalCommon.java:48]: Consider also moving the HTTP status into the method argument so it's clearer *which* status is returned (or rename the method to `throwNotFoundError`?)

File: whois-client/src/main/java/net/ripe/db/whois/api/rest/domain/QueryStrings.java [hunk 8]
```diff
@@ -30,4 +30,13 @@ public QueryStrings() {
     public List<QueryString> getQueryStrings() {
         return queryStrings;
     }
+
+    @Override
+    public String toString() {
+        StringBuilder sb = new StringBuilder();
+        for (QueryString queryString:queryStrings) {
```
[the remark this answers is not in the document]
Comment (MiguelAHM) [reply whois-client/src/main/java/net/ripe/db/whois/api/rest/domain/QueryStrings.java]: removed, it is not necessary

File: whois-client/src/main/java/net/ripe/db/whois/api/rest/domain/QueryStrings.java [hunk 9]
```diff
@@ -30,4 +30,13 @@ public QueryStrings() {
     public List<QueryString> getQueryStrings() {
         return queryStrings;
     }
+
+    @Override
+    public String toString() {
+        StringBuilder sb = new StringBuilder();
```
[the remark this answers is not in the document]
Comment (MiguelAHM) [reply whois-client/src/main/java/net/ripe/db/whois/api/rest/domain/QueryStrings.java]: removed it is not necessary

File: whois-client/src/main/java/net/ripe/db/whois/api/rest/domain/QueryStrings.java [hunk 10]
```diff
@@ -30,4 +30,13 @@ public QueryStrings() {
     public List<QueryString> getQueryStrings() {
         return queryStrings;
     }
+
+    @Override
+    public String toString() {
```
[the remark this answers is not in the document]
Comment (MiguelAHM) [reply whois-client/src/main/java/net/ripe/db/whois/api/rest/domain/QueryStrings.java]: removed it is not necessary

File: whois-api/src/test/java/net/ripe/db/whois/api/rest/WhoisSearchServiceTestIntegration.java (hunk 2 above)
Comment (MiguelAHM) [reply whois-api/src/test/java/net/ripe/db/whois/api/rest/WhoisSearchServiceTestIntegration.java:237]: done, please if there is more required test let me know

File: whois-api/src/main/java/net/ripe/db/whois/api/rest/marshal/AbstractStreamingMarshalCommon.java (hunk 4 above)
Comment (MiguelAHM) [reply whois-api/src/main/java/net/ripe/db/whois/api/rest/marshal/AbstractStreamingMarshalCommon.java]: Done, Thank you! I did not realised

File: whois-api/src/main/java/net/ripe/db/whois/api/rest/RpslObjectStreamer.java (hunk 5 above)
Comment (MiguelAHM) [reply whois-api/src/main/java/net/ripe/db/whois/api/rest/RpslObjectStreamer.java]: Done, I have to take a look at the save actions to do this kind of things manually

File: whois-api/src/main/java/net/ripe/db/whois/api/rest/RestServiceHelper.java (hunk 6 above)
Comment (MiguelAHM) [reply whois-api/src/main/java/net/ripe/db/whois/api/rest/RestServiceHelper.java]: Done

**Review by eshryane — APPROVED** [review]

**Pull request merged**